test/e2e/cache_watch_test.go

```diff
@@ -112,45 +145,47 @@ var _ = Describe("Kubernetes cache watch test:", Label("cache-watch"), func() {
 
 var _ = Describe("Reconciles test:", func() {
 	DescribeTable("Reconciles",
-		func(name string, namespaceIndexesToWatch []int, expectedNamespaceFound []bool) {
+		func(ctx context.Context, testCase *tc) {
 			testData := model.DataProvider(
-				name,
+				fmt.Sprintf("reconcile-%s", CurrentSpecReport().LeafNodeLabels[0]),
 				model.NewEmptyAtlasKeyType().UseDefaultFullAccess(),
 				30010,
 				[]func(*model.TestDataProvider){},
 			)
-			namespaces := setupNamespaces(testData, name, len(expectedNamespaceFound))
-			defer clearNamespaces(testData, namespaces)
-			setupSecrets(testData, namespaces)
-			defer clearSecrets(testData, namespaces)
+			namespaces := setupNamespaces(ctx, testData, testCase.namespaces...)
+			defer clearNamespaces(ctx, testData, namespaces)
+			setupSecrets(ctx, testData, namespaces)
+			defer clearSecrets(ctx, testData, namespaces)
 
-			mgrHandle := setupManager(testData, namespaces, namespaceIndexesToWatch)
-			defer mgrHandle.stop()
+			_, stop := setupManager(ctx, namespaces, testCase.watchNamespaces)
+			defer stop()
 
 			By("Launching an atlas project on each namespace, expect only listened namespaces to update status", func() {
-				projectNames := make([]string, len(expectedNamespaceFound))
+				projectNames := make([]string, len(namespaces))
+				wantToFindSet := sets.NewString(testCase.wantToFind...)
+
 				for i, ns := range namespaces {
 					project := data.DefaultProject()
 					By("Create project", func() {
-						project.Name = utils.RandomName(fmt.Sprintf("test-project-%s-ns%d", name, i))
-						project.Namespace = ns
+						project.Name = ""
+						project.GenerateName = "test-project-" + CurrentSpecReport().LeafNodeLabels[0]
+						project.Namespace = ns.GetName()
 						project.Spec.Name = "" // Set empty name to force validation error
-						projectNames[i] = project.Name
-
 						Expect(testData.K8SClient.Create(context.Background(), project)).ToNot(HaveOccurred())
+						projectNames[i] = project.Name
 					})
 
 					prj := &akov2.AtlasProject{}
 					By("Wait project to be present in Kubernetes", func() {
 						Eventually(func(g Gomega) bool {
 							return g.Expect(
-								testData.K8SClient.Get(testData.Context,
```

Comment:
let's not continue using `testData.Context` (I removed it in my suggested commit). It doesn't allow layering of contexts which is a crucial feature to use (i.e. nested cancellation).